WorkflowUI/Sources/Hosting/WorkflowHostingController.swift

```diff
@@ -22,7 +22,7 @@ import UIKit
 import Workflow
 
 /// Drives view controllers from a root Workflow.
-public final class WorkflowHostingController<ScreenType, Output>: UIViewController where ScreenType: Screen {
+public final class WorkflowHostingController<ScreenType, Output>: WorkflowUIViewController where ScreenType: Screen {
```

Comment:
@square-tomb responding here for threading purposes:

> WorkflowSwiftUI contains a type `WorkflowHostingViewController` that should probably now subclass `WorkflowUIViewController`, too.

good callouts, but given the current motivations for this interface, i think it's okay to punt on the story with SwiftUI and ignore this for the time being.

Reply:
> I wonder, do we need events from`ScreenViewController` at all? Maybe the events from `DescribedViewController` are sufficient if that VC usually maps 1:1 to a screen.

being able to recover the Screen type from ScreenViewController may be of use

> If not, and `DescribedViewController` is often used to display a succession of different screens, we would have to add some events [here](https://github.com/square/workflow-swift/pull/221/files#diff-20be64c4e0d93842089c1503234a2662af08e4a919c9f946e737e94015f5e47bR45-R47). Or we could wrap `currentViewController` in a container `WorkflowUIViewController` if we don't mind deepening the VC hierarchy.

adding some additional events that are more specific to the logic in some of the VC subtypes may be something we want to do eventually, but for now i thought we'd just start with VC lifecycle information and see how far that gets us.